central/authprovider/datastore/datastore_impl.go

```diff
@@ -29,12 +29,31 @@ func (b *datastoreImpl) GetAllAuthProviders(ctx context.Context) ([]*storage.Aut
 	return b.storage.GetAll(ctx)
```

Comment:
I thought we can reuse `GetAuthProvidersFiltered` here, but apparently, the SAC check is skipped here. It's unrelated to changes in this PR, but I think we should actually delegate using the right context to the clients of this datastore, and put a SAC check here. WDYT?

Reply:
You mean `GetAllAuthProviders`, right?

I was wondering the same initially, but decided against it due to the comment specifying that users should need this.
However, looking as its usages being both the `telemetry` as well as the registry, and both specify the right access rights, I changed it to also do a SAC check.